### DISCUSSION TITLE
recv(timeout=2) raising the exception: ord() expected a character, but string of length 0 found.

### DISCUSSION THREAD

**Discussion by ghost** [hardbyte/python-can#1719] [Q&A]
Hi, I'm using a Seeedstudio USB-CANalyzer interface with a bitrate of 500,000. I am communicating with a battery that uses extended arbitration ID. I want to know how long does it take for my battery to enter Sleep Mode (stop sending CAN messages). Hence I am using the next code: 

![image](https://github.com/hardbyte/python-can/assets/140562124/ae91f804-22ab-44fb-9903-ee074e865426)

Currently I have let my script run but randomly I would get the next exception raised.

**Exception has occurred: 
TypeError
ord() expected a character, but string of length 0 found
    line 29, in main
    msg = Bus.recv(timeout=2)
TypeError: ord() expected a character, but string of length 0 found**

The manufacturer comments the time to enter sleep mode, if the battery is not in use, is a bit longer than 2 hours. Altough I could come around with a quick fix or just assume manufacturer is correct, I wonder if there is something I am skipping or I should consider to understand better this error.

Thank you!

**Comment by grant-allan-ctct** [comment]
It's interesting to see the comment [here](https://github.com/hardbyte/python-can/blob/c61981399f5cb479ca52019ca0cd4f076d83da5b/can/interfaces/seeedstudio/seeedstudio.py#L248) that `ser.read` can return an empty string, and yet this possibility is seemingly not catered for in various places, e.g. on lines [259](https://github.com/hardbyte/python-can/blob/c61981399f5cb479ca52019ca0cd4f076d83da5b/can/interfaces/seeedstudio/seeedstudio.py#L259) or [280](https://github.com/hardbyte/python-can/blob/c61981399f5cb479ca52019ca0cd4f076d83da5b/can/interfaces/seeedstudio/seeedstudio.py#L280).
